File: 0049-group-anagrams/0049-group-anagrams.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        vector<bool> seen(strs.size(), false);

        for (size_t i = 0; i < strs.size(); i++) {
            if (!seen[i]) {
                vector<string> temp;
                temp.push_back(strs[i]);
                seen[i] = true;

                for (size_t j = i + 1; j < strs.size(); j++) {
                    if (!seen[j] && isAnagram(strs[i], strs[j])) {
                        temp.push_back(strs[j]);
                        seen[j] = true;
                    }
                }
                ans.push_back(temp);
            }
        }
        return ans;
    }

    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        int arr[26] = {0};
        for (char c : s) {
            arr[c - 'a']++;
        }
        for (char c : t) {
            arr[c - 'a']--;
            if (arr[c - 'a'] < 0) return false;
        }
        return true;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if strs==[""] :return [[""]]
        lst=[''.join(sorted(i)) for i in strs]
        control=list(set(lst))
        if len(control)==0:return []
        ans=[[] for i in range(len(control))]
        for i in range(0,len(strs)):
            if lst[i] in control:ans[control.index(lst[i])].append(strs[i])
        return ans

        
